### PR TITLE
[patch] refactor provider daemon start logic

### PR DESCRIPTION
**Pull request by WindzCUHK**

### CODE REVIEW

**Comment by WindzCUHK** [comment]
@kpango 
passing errors to `ech <-` will cause the force shutdown problem.

1. I removed the `ech <-` inside the errgroup, and add error summary for `providerd`, plx review
2. will do: rename `providerd` to `authorizerD`

**Comment by WindzCUHK** [comment]
blocked by: https://github.com/yahoojapan/athenz-authorizer/pull/32

File: usecase/providerd.go [hunk 1]
```diff
@@ -133,6 +135,7 @@ func (g *providerDaemon) Start(ctx context.Context) <-chan []error {
 func newAuthorizationd(cfg config.Config) (service.Authorizationd, error) {
 	return providerd.New(
 		providerd.WithAthenzURL(cfg.Athenz.URL),
+func newAuthzD(cfg config.Config) (service.Authorizationd, error) {
```
Comment (kevindiu) [inline usecase/providerd.go]: why shorten?

File: usecase/authz_proxyd.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (kevindiu) [inline usecase/authz_proxyd.go]: cannot trace what is changed in this file...

File: usecase/providerd.go (hunk 1 above)
Comment (WindzCUHK) [reply usecase/providerd.go]: go prefer shorter name.
minor change only, you can just revert it.

File: usecase/authz_proxyd.go (hunk 2 above)
Comment (WindzCUHK) [reply usecase/authz_proxyd.go]: not sure why github cannot display after the rename...

ignore last rename commit:
https://github.com/yahoojapan/authorization-proxy/pull/25/files/92d485daaf91fc7a2b2b688962bbd1d586ef5262#diff-098eecd860ad41b7422405e377d9c2a8

**Comment by WindzCUHK** [comment]
update done. pending for review.

**Comment by kevindiu** [comment]
will review after I test on local env.

**Comment by kevindiu** [comment]
Can you revert this?
https://github.com/yahoojapan/authorization-proxy/pull/25/commits/7bd96a1bcaacbdd9a2b92c6f7c69931276cca189

It makes the program hard to read, and the design pattern we can choose to use or not. I can not see any benefit in this use case.

**Comment by kevindiu** [comment]
@WindzCUHK

File: usecase/authz_proxyd.go (hunk 2 above)
Comment (kpango) [reply usecase/authz_proxyd.go]: why don't you use git mv command?

**Comment by WindzCUHK** [comment]
> Can you revert this?
> [7bd96a1](https://github.com/yahoojapan/authorization-proxy/commit/7bd96a1bcaacbdd9a2b92c6f7c69931276cca189)
> 
> It makes the program hard to read, and the design pattern we can choose to use or not. I can not see any benefit in this use case.

The previous one is incorrect anyway.
Do you prefer mutex or other implementation?

File: usecase/authz_proxyd.go (hunk 2 above)
Comment (WindzCUHK) [reply usecase/authz_proxyd.go]: I can see git mv in source tree, just cannot work in github

**Comment by kevindiu** [comment]
LGTM

**Comment by kevindiu** [comment]
Even I don't think it is incorrect, I will delete this logic afterward, so LGTM this time.

**Comment by kevindiu** [comment]
@WindzCUHK Please review this commit and I will merge this PR.
https://github.com/yahoojapan/authorization-proxy/pull/25/commits/06aedbdd6f738aa738eccfc6943ceea8dda9d6c9

**Comment by WindzCUHK** [comment]
related PR: https://github.com/yahoojapan/authorization-proxy/pull/32

**Comment by WindzCUHK** [comment]
1. any other issues left?
2. update `go.mod` after https://github.com/yahoojapan/athenz-authorizer/pull/49 merged?

**Comment by kevindiu** [comment]
@WindzCUHK 
1. no for me.
2. Mayber better to separate 2 release for different changes.

Do u hv any issue left?

File: usecase/authz_proxyd.go (hunk 2 above)
Comment (kevindiu) [reply usecase/authz_proxyd.go]: maybe one thing, can you try to fix it?

Comment (WindzCUHK) [reply usecase/authz_proxyd.go]: fix what?

**Pull request merged**